### PR TITLE
feature: create a package to configure environment variables

### PR DESCRIPTION
**Pull request by matheusTA**
We are having some issues with some contributors, because some time the ENVs are not loading correctly, and it makes the DX while trying to contribute harder and frustrating, so this PR includes a good way to check ENVs, and alert developer about it.

Will be very useful while deploying as well.

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -1,14 +1,22 @@
 {
   "$schema": "https://turbo.build/schema.json",
-  "globalDependencies": ["**/.env.*local"],
-  "globalEnv": ["NEXT_PUBLIC_SOCKET_URL", "NEXT_PUBLIC_SUPABASE_URL", "NEXT_SUPABASE_KEY"],
+  "globalDependencies": [
+    "**/.env.*local"
```
Comment (brunocroh) [inline turbo.json:4]: I didn't know this approach, if I understand it well, you are making the turbo repo get envs directly from the .env file right? 

But I didn't understand why this `*local` here, our example file is ` .env.template` , so maybe it needs to be `**/.env.*template`?

File: packages/env-config/index.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { createEnv } from "@t3-oss/env-nextjs"
+import { z } from "zod"
+
+export const env = createEnv({
+  client: {
+    NEXT_PUBLIC_SOCKET_URL: z.string().default("ws://localhost:4001"),
```
Comment (brunocroh) [inline packages/env-config/index.ts]: The right port is `4000`

```suggestion
    NEXT_PUBLIC_SOCKET_URL: z.string().default("ws://localhost:4000"),
```

File: apps/web/tailwind.config.ts
Comment (brunocroh) [inline apps/web/tailwind.config.ts:1]: This file can be removed from commit, this changes not are related with PR.

File: apps/web/package.json [hunk 3]
```diff
@@ -49,6 +51,7 @@
     "@types/react": "^18.0.22",
     "@types/react-dom": "^18.0.7",
     "autoprefixer": "^10.4.16",
+    "dotenv-cli": "^7.4.2",
```
Comment (brunocroh) [inline apps/web/package.json:49]: Node v20, already has support to load envs from files, maybe over add this library, upgrade to node 20 can be a better option, what do you think?

File: apps/web/package.json [hunk 4]
```diff
@@ -3,9 +3,10 @@
   "version": "1.0.0",
   "private": true,
   "scripts": {
-    "dev": "next dev",
-    "build": "next build",
-    "start": "next start",
+    "env:load": "dotenv -e ../../.env --",
```
Comment (brunocroh) [inline apps/web/package.json:6]: I believe searching for parent folders is bad practice, perhaps this could be done in the root repo in a pipeline step in the turborepo.

File: apps/web/lib/supabase/server.ts [hunk 5]
```diff
@@ -7,8 +8,8 @@ export function createClient(): SupabaseClient<Database> {
   const cookieStore = cookies()
 
   return createServerClient<Database>(
-    process.env.NEXT_PUBLIC_SUPABASE_URL!,
-    process.env.NEXT_SUPABASE_KEY!,
+    env.NEXT_PUBLIC_SUPABASE_URL,
+    env.NEXT_PUBLIC_SUPABASE_ANON_KEY,
```
Comment (brunocroh) [inline apps/web/lib/supabase/server.ts]: This key is not public, and not an anon key from supabase.

So the previous name are better, to next not expose it, and to represent the right key on supabase, please revert it to `NEXT_SUPABASE_KEY`

File: apps/web/app/permissions.d.ts
Comment (brunocroh) [inline apps/web/app/permissions.d.ts:1]: This file is not related with changes on PR, please remove this file from commit.

**Review by brunocroh — CHANGES_REQUESTED** [review]
Firstly, thank you a lot, I really appreciate this contribution.

I just did some comments, that we need to solve before merge it.

Awesome work dude!

File: turbo.json (hunk 1 above)
Comment (matheusTA) [reply turbo.json:4]: This is a default turborepo configuration, I just removed the `globalEnv` and when I saved the file the lint adjusted the `globalDependencies` indentation

Comment (brunocroh) [reply turbo.json:4]: Oh, sure, sorry.

But why remove globalEnv?

File: apps/web/lib/supabase/server.ts (hunk 5 above)
Comment (matheusTA) [reply apps/web/lib/supabase/server.ts]: Next environment variables must have the prefix `NEXT_PUBLIC`, we will get type errors if these are not prefixed.

We can put `NEXT_PUBLIC_SUPABASE_KEY`

File: turbo.json (hunk 1 above)
Comment (matheusTA) [reply turbo.json:4]: I thought we wouldn't need it since I was using the package env, but I saw that in the package we received a warning when removing the `globalEnv`. 

so I'll put it back

Comment (brunocroh) [reply turbo.json:4]: Yes sure, how we handle it on library, is totally useless to repeat it here.

nice, thank you.

Comment (brunocroh) [reply turbo.json:4]: looking in the turboRepo docs, I see this alternative ` "globalDotEnv": [".env"],`
Maybe it will fit better our usecase.

Comment (brunocroh) [reply turbo.json:4]: https://turbo.build/repo/docs/handbook/environment-variables

File: apps/web/lib/supabase/server.ts (hunk 5 above)
Comment (brunocroh) [reply apps/web/lib/supabase/server.ts]: `NEXT_PUBLIC` is when you want to expose this envs to the browser.

In this case, this env cannot be public, because is really sensitive data.

So because of that it not have `NEXT_PUBLIC` prefix, just `NEXT`, to we know this is used on Vercel cloud.

I running it with dev command
Screenshot 2024-06-02 at 20 18 31

and running build
Screenshot 2024-06-02 at 20 19 00

and not have any warnings about it on my side.
If I skipping something let me know please.

But we cannot add `NEXT_PUBLIC` on this one, the env `NEXT_SUPABASE_KEY` is just consumed on server action. If it is used outside of server side, need to throw an error.

Comment (matheusTA) [reply apps/web/lib/supabase/server.ts]: I got it, thanks.

I was putting them as client env so I was getting a warning. I put them as server env and it worked

File: turbo.json (hunk 1 above)
Comment (matheusTA) [reply turbo.json:4]: Yes, it works perfectly

File: apps/web/lib/supabase/server.ts (hunk 5 above)
Comment (brunocroh) [reply apps/web/lib/supabase/server.ts]: Great!

File: apps/web/package.json (hunk 4 above)
Comment (matheusTA) [reply apps/web/package.json:6]: We need to load the env here, any suggestions on how to do it in the root repo?

File: apps/web/package.json (hunk 3 above)
Comment (matheusTA) [reply apps/web/package.json:49]: Updating the node can break flows outside the scope of this PR, is it worth it?

Comment (brunocroh) [reply apps/web/package.json:49]: Yes, you are right, we can follow with these changes, and try this approach with node v20 in a new PR.

File: apps/web/package.json (hunk 4 above)
Comment (brunocroh) [reply apps/web/package.json:6]: If we move this command `env:load`, to root `package.json`, and run it on turbo repo pipeline, I believe it can work, but I never try it, what do you think?

I'm not had a huge experience using turbo repo pipelines, but it makes sense for me, if it is not possible please let me know.

And once again, thank you, you brought a lot of knowledge and good approaches with this PR.

Comment (brunocroh) [reply apps/web/package.json:6]: If you believe is better try it on onther PR, let me know, we can continue with it, dont worry.

File: pnpm-lock.yaml
Comment (brunocroh) [inline pnpm-lock.yaml:1]: I need you remove changes on this file to merge it

File: apps/web/package.json (hunk 4 above)
Comment (matheusTA) [reply apps/web/package.json:6]: I also don't have much experience with turborepo, I think we can move on and try in another PR

**Review by brunocroh — APPROVED** [review]
